### PR TITLE
Chat gpt feature

### PR DESCRIPTION
**Pull request by Axlfc**
#134

### CODE REVIEW

File: data/features/chatGPT/bot.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+            max_tokens = 1024,
+            n = 1,
+            temperature = 0.5,
+        )
+
+        response = completions.choices[0].text
+        return response
+
+
+def main():
```
Comment (AleixMT) [inline data/features/chatGPT/bot.py:48]: Keep Only one main defined

File: data/features/chatGPT/chatGPT.func.sh [hunk 2]
```diff
@@ -0,0 +1,11 @@
+install_chatGPT_mid() {
+  if [ ! -f "${BIN_FOLDER}/chatGPT/conversations" ]; then
+    mkdir "${BIN_FOLDER}/chatGPT/conversations"
+  fi
+  rm "${XDG_DOCUMENTS_DIR}/chatGPT_conversations"
+  ln -s "${BIN_FOLDER}/chatGPT/conversations" "${XDG_DOCUMENTS_DIR}/chatGPT_conversations"
+}
+
+uninstall_chatGPT_mid() {
+  rm "${XDG_DOCUMENTS_DIR}/chatGPT_conversations"
```
Comment (AleixMT) [inline data/features/chatGPT/chatGPT.func.sh:10]: use an api function

File: data/features/chatGPT/chatGPT.func.sh [hunk 3]
```diff
@@ -0,0 +1,11 @@
+install_chatGPT_mid() {
+  if [ ! -f "${BIN_FOLDER}/chatGPT/conversations" ]; then
+    mkdir "${BIN_FOLDER}/chatGPT/conversations"
+  fi
+  rm "${XDG_DOCUMENTS_DIR}/chatGPT_conversations"
+  ln -s "${BIN_FOLDER}/chatGPT/conversations" "${XDG_DOCUMENTS_DIR}/chatGPT_conversations"
```
Comment (AleixMT) [inline data/features/chatGPT/chatGPT.func.sh:6]: use api

File: data/features/chatGPT/chatGPT.func.sh [hunk 4]
```diff
@@ -0,0 +1,11 @@
+install_chatGPT_mid() {
+  if [ ! -f "${BIN_FOLDER}/chatGPT/conversations" ]; then
+    mkdir "${BIN_FOLDER}/chatGPT/conversations"
```
Comment (AleixMT) [inline data/features/chatGPT/chatGPT.func.sh:3]: use api

File: data/features/chatGPT/chatGPT.sh [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+chatGPT()
+{
+  if [ ! -f "€{BIN_FOLDER}/chatGPT/.env" ]; then
+    echo "You need to enter a valid API key in the file €{BIN_FOLDER}/chatGPT/.env"
+    echo "Go to https://beta.openai.com/account/api-keys to get yours."
+    echo "Edit or create the .env file and write inside the file:"
+    echo "OPENAI_API_KEY=your_open_ai_api_key_code"
+  else
+    if [ ! "$1" ]; then
```
Comment (AleixMT) [inline data/features/chatGPT/chatGPT.sh:12]: what is the condition that you are checking in the if? [ ! "string" ]   ??

Rewrite it with the evaluations that we use in the customizer, this is not how we check the number of args

File: data/features/chatGPT/chatgpt_exec.sh [hunk 6]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env bash
+if [ ! -f "€{BIN_FOLDER}/chatGPT/.env" ]; then
+  echo "You need to enter a valid API key in the file €{BIN_FOLDER}/chatGPT/.env"
+  echo "Go to https://beta.openai.com/account/api-keys to get yours."
+  echo "Edit or create the .env file and write inside the file:"
+  echo "OPENAI_API_KEY=your_open_ai_api_key_code"
+  read error_val
+else
```
Comment (AleixMT) [inline data/features/chatGPT/chatgpt_exec.sh:8]: .env present? check

OPENAI_API_KEY defined? no check. Check execution conditiion before entering code. use guard clauses.

**Review by AleixMT — CHANGES_REQUESTED** [review]
not apt to customizer standards

File: data/features/chatGPT/bot.py (hunk 1 above)
Comment (Axlfc) [reply data/features/chatGPT/bot.py:48]: Executing the main() function:

```
if __name__ == '__main__':
    main()
```

This code ensures that the main() function is only executed when the script is run directly and not imported as a module.

**Comment by AleixMT** [comment]
@Axlfc

**Comment by AleixMT** [comment]
Since you are not addressing the requested changes in half a year 
image
 this branch will be deleted.

**Pull request closed without merging**